capsules/src/qdec.rs

```diff
@@ -0,0 +1,108 @@
+//! Provides userspace access to the Qdec on a board.
+//!
+//! Usage
+//! -----
+//!
+//! ````
+//! let qdec = static_init!(
+//!     capsules::qdec::Qdec<'static>,
+//!     capsules::qdec::QdecInterface::new(&nrf52::qdec::QDEC,
+//!                                         kernel::Grant::create())
+//! );
+//! kernel::hil::QdecDriver.set_client(qdec);
+//! ````
+//! #Interrupt Spurred Readings versus Regular Readings
+//! An application can either enable interrupts to get the
+//! accumulation value or manually read it whenever it wants
+
+use crate::driver;
+use kernel::hil;
+use kernel::{AppId, Callback, Driver, Grant, ReturnCode};
+
+pub const DRIVER_NUM: usize = driver::NUM::Qdec as usize;
+
+/// This struct contains the resources necessary for the QdecInterface
+pub struct QdecInterface<'a> {
+    driver: &'a dyn hil::qdec::QdecDriver,
+    apps: Grant<App>,
+}
+
+#[derive(Default)]
+/// This struct contains the necessary fields for an app
+pub struct App {
+    callback: Option<Callback>,
+    pos: u32,
```

Comment:
```suggestion
    position: u32,
```

Generally, Tock style prefers spelled out variable names.

---

A more semantic question, should this be called `position` or `ticks`? Generally position has some tie-in to physical space and implies an absolute position. If a wheel goes forward for a long time, it's meaningful for a number of encoder ticks to wraparound, it's not really meaningful for a position to wrap.

Reply:
Thanks for the query! The value returned is the absolute position relative to the starting location of the rotary encoder. Currently, the value we return does not wrap if the rotary encoder makes a full rotation-it continues ticking up or down (depending on the direction). Do you think this should be updated to cater more to `ticks`?